pkg/util/discovery/server_resourcer_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package discovery
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
```

Comment:
I know we use testify elsewhere, but I think this is our first use of mock.Mock. Everywhere else we're mostly using custom fakes. I don't want to make this more work, but I also don't really want to start a precedent of using mocks without discussing it with the team first. Using Mocks than get out of hand pretty quick. That said, it does look a little simpler than writing a fake.

Reply:
Changed to a custom mock in the latest patch.